File: programmers/level2/프린터.java
package level2;

import java.util.LinkedList;
import java.util.Queue;

public class 프린터 {

	public static int solution(int[] priorities, int location) {
		int answer = 0;
		int[] count = new int[10]; // 우선순위 별 개수를 넣음
		int max = 0;
		Queue<Integer> queue = new LinkedList<Integer>();

		// 큐에 데이터를 순서대로 넣음
		for (int i : priorities) {
			max = max < i ? i : max;
			count[i]++;
			queue.add(i);
		}

		while (true) {
			int n = queue.poll();
			location--;
			if (n == max) {
				answer++;
				if (location == -1) {
					break;
				}
				count[max]--;
				for (int i = max; i > 0; i--) {
					if (count[i] > 0) {
						max = i;
						break;
					}
				}
			} else {
				if (location == -1) {
					location = queue.size();
				}
				queue.add(n);
			}
		}

		return answer;
	}

	public static void main(String[] args) {
//		System.out.println(solution(new int[] { 2, 1, 3, 2 }, 2));
		System.out.println(solution(new int[] { 1, 1, 9, 1, 1, 1 }, 0));
	}
}

File: programmers/level2/프린터.py
def solution(priorities, location):
    answer = 0

    while len(priorities) > 0:
        if priorities[0] == max(priorities):  # 우선순위 확인
            # 맨 앞 문서 인쇄
            priorities.pop(0)
            answer += 1
            if location == 0:  # 요청한 문서이면 반복문 종료
                break
        else:
            # 맨 앞 문서를 대기 목록의 가장 마지막에 추가
            priorities.append(priorities.pop(0))
        location = location - 1 if location > 0 else len(priorities) - 1  # 요청 문서 위치 변경
    return answer


# solution([2, 1, 3, 2], 2)
solution([1, 1, 9, 1, 1, 1], 0)
# solution([9, 1, 9, 1, 2, 1], 3)
